src/libasr/pass/replace_symbolic.cpp

```diff
@@ -0,0 +1,764 @@
+                nullptr, nullptr, ASR::storage_typeType::Default, ASRUtils::TYPE(ASR::make_CPtr_t(al, loc)),
+                nullptr, ASR::abiType::BindC, ASR::Public, ASR::presenceType::Required, true));
+            fn_symtab->add_symbol(s2c(al, "x"), arg1);
+            args.push_back(al, ASRUtils::EXPR(ASR::make_Var_t(al, loc, arg1)));
+            ASR::symbol_t* arg2 = ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(
+                al, loc, fn_symtab, s2c(al, "y"), nullptr, 0, ASR::intentType::In,
+                nullptr, nullptr, ASR::storage_typeType::Default, ASRUtils::TYPE(ASR::make_CPtr_t(al, loc)),
+                nullptr, ASR::abiType::BindC, ASR::Public, ASR::presenceType::Required, true));
+            fn_symtab->add_symbol(s2c(al, "y"), arg2);
+            args.push_back(al, ASRUtils::EXPR(ASR::make_Var_t(al, loc, arg2)));
+            ASR::symbol_t* arg3 = ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(
+                al, loc, fn_symtab, s2c(al, "z"), nullptr, 0, ASR::intentType::In,
+                nullptr, nullptr, ASR::storage_typeType::Default, ASRUtils::TYPE(ASR::make_CPtr_t(al, loc)),
+                nullptr, ASR::abiType::BindC, ASR::Public, ASR::presenceType::Required, true));
+            fn_symtab->add_symbol(s2c(al, "z"), arg3);
+            args.push_back(al, ASRUtils::EXPR(ASR::make_Var_t(al, loc, arg3)));
+
+            Vec<ASR::stmt_t*> body;
+            body.reserve(al, 1);
+
+            Vec<char*> dep;
+            dep.reserve(al, 1);
+
+            ASR::asr_t* new_subrout = ASRUtils::make_Function_t_util(al, loc,
+                fn_symtab, s2c(al, new_name), dep.p, dep.n, args.p, args.n, body.p, body.n,
+                nullptr, ASR::abiType::BindC, ASR::accessType::Public,
+                ASR::deftypeType::Interface, s2c(al, new_name), false, false, false,
+                false, false, nullptr, 0, false, false, false, s2c(al, header));
+            ASR::symbol_t* new_symbol = ASR::down_cast<ASR::symbol_t>(new_subrout);
+            module_scope->add_symbol(s2c(al, new_name), new_symbol);
+        }
+
+        ASR::symbol_t* func_sym = module_scope->get_symbol(new_name);
+        Vec<ASR::call_arg_t> call_args;
+        call_args.reserve(al, 3);
+        ASR::call_arg_t call_arg1, call_arg2, call_arg3;
+        call_arg1.loc = loc;
+        call_arg1.m_value = value1;
+        call_arg2.loc = loc;
+        if (ASR::is_a<ASR::IntrinsicFunction_t>(*value2)){
+            ASR::IntrinsicFunction_t *s = ASR::down_cast<ASR::IntrinsicFunction_t>(value2);
+            this->visit_IntrinsicFunction(*s);
+        }
+        call_arg2.m_value = value2;
+        call_arg3.loc = loc;
+        if (ASR::is_a<ASR::IntrinsicFunction_t>(*value3)){
+            ASR::IntrinsicFunction_t *s = ASR::down_cast<ASR::IntrinsicFunction_t>(value3);
+            this->visit_IntrinsicFunction(*s);
+        }
+        call_arg3.m_value = value3;
+        call_args.push_back(al, call_arg1);
```

Comment:
We just need to edit call_arg2.m_value and call_arg3.m_value , so that we can go from . 
```
(SubroutineCall
                                        2 basic_add
                                        2 basic_add
                                        [((Var 3 x))
                                        ((IntrinsicFunction
                                            SymbolicPi
                                            []
                                            0
                                            (SymbolicExpression)
                                            ()
                                        ))
                                        ((IntrinsicFunction
                                            SymbolicSymbol
                                            [(StringConstant
                                                "x"
                                                (Character 1 1 ())
                                            )]
                                            0
                                            (SymbolicExpression)
                                            ()
                                        ))]
                                        ()
                                    )
```
To 
```
(SubroutineCall
                                        2 basic_add
                                        2 basic_add
                                        [((Var 3 x))
                                         ((Var 3 queue0))
                                         ((Var 3 queue1))]
                                        ()
                                    )
```
I'm not sure how this has to be done , but once this is done ,we should be able to get an output

Reply:
You can use `ASR::stmt _t *tmp` global variable to store the queue0 in the visit_IntrinsicFunction and access that in the `perform_symbolic_binary_operation`.